CMakeLists.txt

```diff
@@ -72,13 +82,10 @@ ELSE()
   ELSE()
     ADD_COMPILE_FLAG("-fPIC")
   ENDIF()
-  IF(uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
-    ADD_COMPILE_FLAG("-O0")
-    ADD_COMPILE_FLAG("-g3")
-  ELSE()
-    ADD_COMPILE_FLAG("-O2")
-    ADD_DEFINITIONS("-UNDEBUG")  # Keep asserts.
-  ENDIF()
+  ADD_DEBUG_COMPILE_FLAG("-O0")
+  ADD_DEBUG_COMPILE_FLAG("-g3")
+  ADD_NONDEBUG_COMPILE_FLAG("-O2")
```

Comment:
Interestingly I just checked and it looks like CMake adds '-O3' in release mode and '-g' in debug mode automatically, and these appear on the command line *after* (and thus override) our manually-specified flags. So maybe they aren't necessary anyway? (OTOH the `-DUNDEBUG` does appear after the automatically-added `-DNDEBUG` and does have the desired effect).